internal/factsengine/gatherers/saplocalhostresolver_test.go

```diff
@@ -0,0 +1,154 @@
+	suite.NoError(err)
+
+	suite.mockResolver.On("LookupHost", "sapqasas").Return([]string{"10.1.1.5"}, nil)
+	suite.mockPinger.On("Ping", "sapqasas").Return(true, nil)
+	suite.mockResolver.On("LookupHost", "sapnwper").Return([]string{"10.1.1.6"}, nil)
+	suite.mockPinger.On("Ping", "sapnwper").Return(false, nil)
+
+	g := gatherers.NewSapLocalhostResolver(appFS, suite.mockResolver, suite.mockPinger)
+
+	factRequests := []entities.FactRequest{{
+		Name:     "sap_localhost_resolver",
+		Gatherer: "sap_localhost_resolver",
+		CheckID:  "check1",
+	}}
+
+	factResults, err := g.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:    "sap_localhost_resolver",
+			CheckID: "check1",
+			Value: &entities.FactValueMap{
+				Value: map[string]entities.FactValue{
+					"QAS": &entities.FactValueList{
+						Value: []entities.FactValue{
+							&entities.FactValueMap{
+								Value: map[string]entities.FactValue{
+									"hostname": &entities.FactValueString{Value: "sapqasas"},
+									"addresses": &entities.FactValueList{
+										Value: []entities.FactValue{
+											&entities.FactValueString{Value: "10.1.1.5"},
+										},
+									},
+									"instance_name": &entities.FactValueString{Value: "ASCS00"},
+									"reachability":  &entities.FactValueBool{Value: true},
+								},
+							},
+						},
+					},
+					"NWP": &entities.FactValueList{
+						Value: []entities.FactValue{
+							&entities.FactValueMap{
+								Value: map[string]entities.FactValue{
+									"hostname": &entities.FactValueString{Value: "sapnwper"},
+									"addresses": &entities.FactValueList{
+										Value: []entities.FactValue{
+											&entities.FactValueString{Value: "10.1.1.6"},
+										},
+									},
+									"instance_name": &entities.FactValueString{Value: "ER10"},
+									"reachability":  &entities.FactValueBool{Value: false},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	suite.NoError(err)
+	suite.Equal(expectedResults, factResults)
+}
+
+func (suite *SapLocalhostResolverTestSuite) TestSapLocalhostResolverNoProfiles() {
+	appFS := afero.NewMemMapFs()
+
+	err := appFS.MkdirAll("/usr/sap/QAS", 0644)
+	suite.NoError(err)
+
+	g := gatherers.NewSapLocalhostResolver(appFS, suite.mockResolver, suite.mockPinger)
+
+	factRequests := []entities.FactRequest{{
+		Name:     "sap_localhost_resolver",
+		Gatherer: "sap_localhost_resolver",
+		CheckID:  "check1",
+	}}
+
+	factResults, err := g.Gather(factRequests)
+	suite.Nil(factResults)
+	suite.EqualError(err, "fact gathering error: saplocalhost_resolver-file-system-error - error reading the sap profiles file system: open /sapmnt/QAS/profile: file does not exist")
+}
+
+func (suite *SapLocalhostResolverTestSuite) TestSapLocalhostResolverLookupHostError() {
+	appFS := afero.NewMemMapFs()
+
+	err := appFS.MkdirAll("/usr/sap/QAS", 0644)
+	suite.NoError(err)
+
+	ascsProfileFile, _ := os.Open(helpers.GetFixturePath("gatherers/sap_profile.ascs"))
```

Comment:
I don't think we need to get the content of a file. We don't use it.
```
err = afero.WriteFile(appFS, "/sapmnt/QAS/profile/QAS_ASCS00_sapqasas", []byte{}, 0644)
```
works for us